src/Server.php

```diff
@@ -28,10 +29,15 @@ public function __construct($uri, LoopInterface $loop, array $context = array())
             $scheme = substr($uri, 0, $pos);
         }
 
-        $server = new TcpServer(str_replace('tls://', '', $uri), $loop, $context['tcp']);
+        if ($scheme === 'unix') {
+            $server = new UnixServer($uri, $loop, $context['unix']);
+        }
+        else {
+            $server = new TcpServer(str_replace('tls://', '', $uri), $loop, $context['tcp']);
 
-        if ($scheme === 'tls') {
-            $server = new SecureServer($server, $loop, $context['tls']);
+            if ($scheme === 'tls') {
+                $server = new SecureServer($server, $loop, $context['tls']);
+            }
```

Comment:
Maybe export the Server constructing logic to a separate method like this: https://github.com/the-eater/socket/commit/2defb1b146f85dcd28510d725ce7b7aece7c644d?diff=split

Reply:
Functionally LGTM, however I don't currently see this covered in the test suite (plus minor CS issue above). Can you update this? :shipit: :+1:

Reply:
@clue what's your expectation here? Testing the type of the create private server member? Would need a hint how to do this due to the private nature.

Reply:
This class is a facade that mostly contains some glue code to make the underlying APIs more easily accessible. Just check the other high-level functional tests and the code coverage. IMHO it's sufficient to create an Unix domain socket server with this facade and check a client connection can be established.

Reply:
Gotcha. Tests added.